crates/client/src/lib.rs

```diff
@@ -33,63 +35,117 @@ use rustls::kx_group::{SECP256R1, SECP384R1, X25519};
 use rustls::version::TLS13;
 use rustls::{Certificate, OwnedTrustAnchor, PrivateKey, RootCertStore};
 
+mod private {
+    pub trait Scope: Copy + Clone {}
+}
+
+pub trait Scope: private::Scope {}
```

Comment:
Why have a "private" `Scope` just to re-declare it as `pub`?

Reply:
https://rust-lang.github.io/api-guidelines/future-proofing.html#sealed-traits-protect-against-downstream-implementations-c-sealed

Reply:
This means that only we can implement this trait, essentially I am modelling an enum here